[Bug]: Build failed installing from cargo

### What did you expect to happen?

I built from source via `cargo install atuin` on my raspberry pi.
I expected atuin to compile and install successfully.
Machine details:
Raspbian GNU/Linux 11 (bullseye) aarch64
Raspberry Pi 4 Model B Rev 1.2

### What happened?

Compilation failed with the following:
```
   Compiling atuin-client v18.4.0
error[E0283]: type annotations needed
  --> /home/yucjo/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/atuin-client-18.4.0/src/sync.rs:89:52
   |
89 |         if history.len() < remote_status.page_size.try_into().unwrap() {
   |                          -                         ^^^^^^^^
   |                          |
   |                          type must be known at this point
   |
   = note: multiple `impl`s satisfying `usize: PartialOrd<_>` found in the following crates: `core`, `deranged`:
           - impl PartialOrd for usize;
           - impl<MIN, MAX> PartialOrd<deranged::RangedUsize<MIN, MAX>> for usize
             where the constant `MIN` has type `usize`, the constant `MAX` has type `usize`;
help: try using a fully qualified path to specify the expected types
   |
89 |         if history.len() < <i64 as TryInto<T>>::try_into(remote_status.page_size).unwrap() {
   |                            ++++++++++++++++++++++++++++++                       ~
```

### Atuin doctor output

```yaml
Does not apply
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/crates/atuin-client/src/sync.rs
+++ b/crates/atuin-client/src/sync.rs
@@ -85,8 +85,9 @@ async fn sync_download(
         db.save_bulk(&history).await?;
 
         local_count = db.history_count(true).await?;
+        let remote_page_size = std::cmp::max(remote_status.page_size, 0) as usize;
 
-        if history.len() < remote_status.page_size.try_into().unwrap() {
+        if history.len() < remote_page_size {
             break;
         }
 

--- a/crates/atuin-client/src/sync.rs
+++ b/crates/atuin-client/src/sync.rs
@@ -1,5 +1,4 @@
 use std::collections::HashSet;
-use std::convert::TryInto;
 use std::iter::FromIterator;
 
 use eyre::Result;

